docs/source/onnx.rst

```diff
@@ -1,745 +1,64 @@
+`ONNX graph <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
 
-The process for adding a symbolic function depends on the type of operator.
+The exported model can be consumed by any of the many
+`runtimes that support ONNX <https://onnx.ai/supported-tools.html#deployModel>`_, including
+Microsoft's `ONNX Runtime <https://www.onnxruntime.ai>`_.
 
-.. _adding-support-aten:
+**There are two flavors of ONNX exporter API that you can use, as listed below:**
 
-ATen operators
-^^^^^^^^^^^^^^
+TorchDynamo-based ONNX Exporter
+-------------------------------
 
-`ATen <https://pytorch.org/cppdocs/#aten>`_ is PyTorch's built-in tensor library.
-If the operator is an ATen operator (shows up in the TorchScript graph with the prefix
-``aten::``), make sure it is not supported already.
+*The TorchDynamo-based ONNX exporter is the newest (still and beta) **exporter for PyTorch 2.0 and newer**.
 
-List of supported operators
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+TorchDynamo engine is leveraged to hook into Python's frame evaluation API and dynamically rewrite its
+bytecode into an FX Graph. The resulting FX Graph is then polished before it is finally translated into an
+ONNX graph.
 
-Visit the auto generated :doc:`list of supported TorchScript operators <../onnx_supported_aten_ops>`
-for details on which operator are supported in each ``opset_version``.
+The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>`_ is captured using
+bytecode analysis that preserves the dynamic nature of the model instead of using traditional static tracing techniques.
 
-Adding support for an aten or quantized operator
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:doc:`Learn more about the TorchDynamo-based ONNX Exporter <onnx_dynamo>`
 
-If the operator is not in the list above:
+TorchScript-based ONNX Exporter
+-------------------------------
 
-* Define the symbolic function in ``torch/onnx/symbolic_opset<version>.py``, for example
-  `torch/onnx/symbolic_opset9.py <https://github.com/pytorch/pytorch/blob/main/torch/onnx/symbolic_opset9.py>`_.
-  Make sure the function has the same name as the ATen function, which may be declared in
-  ``torch/_C/_VariableFunctions.pyi`` or ``torch/nn/functional.pyi`` (these files are generated at
-  build time, so will not appear in your checkout until you build PyTorch).
-* By default, the first arg is the ONNX graph.
-  Other arg names must EXACTLY match the names in the ``.pyi`` file,
-  because dispatch is done with keyword arguments.
-* In the symbolic function, if the operator is in the
-  `ONNX standard operator set <https://github.com/onnx/onnx/blob/master/docs/Operators.md>`_,
-  we only need to create a node to represent the ONNX operator in the graph.
-  If not, we can compose several standard operators that have the
-  equivalent semantics to the ATen operator.
+*The TorchScript-based ONNX exporter is available since PyTorch 1.2.0*
```

Comment:
not important nor do I remember the specifics, but it was available way before 1.2.0, probably 0.3.0

Reply:
Yeah I was in doubt whether to use 0.3.0 or 1.2.0

the `torch.onnx` was indeed introduced at 0.3.0, but according to the [1.2.0 release notes](https://github.com/pytorch/pytorch/releases/tag/v1.2.0) in `PyTorch 1.2, working with Microsoft, we’ve added full support to export ONNX Opset versions 7(v1.2), 8(v1.3), 9(v1.4) and 10 (v1.5).` which I assumed to be the official non-caffe2 onnx support. I can change to 0.3.0 too

Reply:
My impression was that the TorchScript support came in this release too